packages/runed/src/lib/functions/box/box.svelte.ts

```diff
@@ -0,0 +1,194 @@
+		[BoxSymbol]: true,
+		[isWritableSymbol]: true,
+		get value() {
+			return value as unknown;
+		},
+		set value(v: unknown) {
+			value = v;
+		},
+	};
+}
+
+/**
+ * Creates a readonly box
+ *
+ * @param getter Function to get the value of the box
+ * @returns A box with a `value` property whose value is the result of the getter.
+ * Useful to pass state to other functions.
+ */
+function boxWith<T>(getter: () => T): ReadableBox<T>;
+/**
+ * Creates a writable box
+ *
+ * @param getter Function to get the value of the box
+ * @param setter Function to set the value of the box
+ * @returns A box with a `value` property which can be set to a new value.
+ * Useful to pass state to other functions.
+ */
+function boxWith<T>(getter: () => T, setter: (v: T) => void): WritableBox<T>;
+function boxWith<T>(getter: () => T, setter?: (v: T) => void) {
+	const derived = $derived.by(getter);
+
+	if (setter) {
+		return {
+			[BoxSymbol]: true,
+			[isWritableSymbol]: true,
+			get value() {
+				return derived;
+			},
+			set value(v: T) {
+				setter(v);
+			},
+		};
+	}
+
+	return {
+		[BoxSymbol]: true,
+		get value() {
+			return getter();
+		},
+	};
+}
+
+export type BoxFrom<T> =
+	T extends WritableBox<infer U>
+		? WritableBox<U>
+		: T extends ReadableBox<infer U>
+			? ReadableBox<U>
+			: T extends Getter<infer U>
+				? ReadableBox<U>
+				: WritableBox<T>;
+
+/**
+ * Creates a box from either a static value, a box, or a getter function.
+ * Useful when you want to receive any of these types of values and generate a boxed version of it.
+ *
+ * @returns A box with a `value` property whose value.
+ */
+function boxFrom<T>(value: T): BoxFrom<T> {
+	if (box.isBox(value)) return value as BoxFrom<T>;
+	if (isFunction(value)) return boxWith(value) as BoxFrom<T>;
+	return box(value) as BoxFrom<T>;
+}
+
+type GetKeys<T, U> = {
+	[K in keyof T]: T[K] extends U ? K : never;
+}[keyof T];
+type RemoveValues<T, U> = Omit<T, GetKeys<T, U>>;
+
+type BoxFlatten<R extends Record<string, unknown>> = Expand<
+	RemoveValues<
+		{
+			[K in keyof R]: R[K] extends WritableBox<infer T> ? T : never;
+		},
+		never
+	> &
+		RemoveValues<
+			{
+				readonly [K in keyof R]: R[K] extends WritableBox<infer _>
+					? never
+					: R[K] extends ReadableBox<infer T>
+						? T
+						: never;
+			},
+			never
+		>
+> &
+	RemoveValues<
+		{
+			[K in keyof R]: R[K] extends ReadableBox<infer _> ? never : R[K];
+		},
+		never
+	>;
+
+/**
+ * Function that gets an object of boxes, and returns an object of reactive values
+ *
+ * @example
+ * const count = box(0)
+ * const flat = box.flatten({ count, double: box.with(() => count.value) })
+ * // type of flat is { count: number, readonly double: number }
+ */
+function boxFlatten<R extends Record<string, unknown>>(boxes: R): BoxFlatten<R> {
+	return Object.entries(boxes).reduce<BoxFlatten<R>>((acc, [key, b]) => {
+		if (!box.isBox(b)) {
+			return Object.assign(acc, { [key]: b });
+		}
+
+		const value = $derived(b.value);
```

Comment:
Hey, minor note, the `$derived` here is unnecessary because any computed value is already wrapped with a `derived`. See `both..with`.